### PR TITLE
fluentd log split - update with nil check

### PR DESCRIPTION
**Pull request by mjudeikis**
```release-note
Split kubernetes containers logs in geneva
```

### CODE REVIEW

File: pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml [hunk 1]
```diff
@@ -22,6 +22,17 @@ data:
         tag journald
       </rule>
     </match>
+    <filter journald>
+      @type record_transformer
+      enable_ruby true
+      <record>
+       #k8s_apiserver_apiserver-vqxg4_kube-service-catalog_72ce4d73-5224-11e9-98d0-000d3a196756_0
+       CONTAINER ${unless record["CONTAINER_NAME"].nil? ; record["CONTAINER_NAME"].split("_")[1] ; else "" ; end; }
+       POD ${unless record["CONTAINER_NAME"].nil?; record["CONTAINER_NAME"].split("_")[2] ; else "" ; end; }
```
Comment (mjudeikis) [inline pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: `else ""` needed if we dont want to see `Null` values in Geneva. an empty string value is a default for all other empty fields.

File: pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml [hunk 2]
```diff
@@ -22,6 +22,17 @@ data:
         tag journald
       </rule>
     </match>
+    <filter journald>
+      @type record_transformer
+      enable_ruby true
+      <record>
+       #k8s_apiserver_apiserver-vqxg4_kube-service-catalog_72ce4d73-5224-11e9-98d0-000d3a196756_0
+       CONTAINER ${unless record["CONTAINER_NAME"].nil? ; record["CONTAINER_NAME"].split("_")[1] ; else "" ; end; }
```
Comment (ehashman) [inline pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: Is this 1-indexed or is the [0] field special? I guess I don't actually know what the source data looks like.

Comment (mjudeikis) [reply pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: data format is comment field above. so first field [0] is `k8s_`

Comment (mjudeikis) [reply pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: comment is not very informative :/

Comment (ehashman) [reply pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: Ahh, makes sense. Maybe we should update the comment to be clearer? :)

File: pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml (hunk 1 above)
Comment (ehashman) [reply pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: Out of curiosity, why do we prefer an empty string to Null here?

Comment (mjudeikis) [reply pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: my preference is to see a unified view in UI, not some fields `""` some `Null` . this way fields are either populated or `""`
 view now is empty fields if field is not populated 
image

I dont have a hard preference :man_shrugging:

Comment (jim-minter) [reply pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: empty string is not the default, nil is - please fix

**Comment by jim-minter** [comment]
@mjudeikis I'd like #1455 to get in as a priority as it adds crash dump logging, and the 2 PRs will conflict.  Please rebase this (please also consider using the ternary operator like I do for MESSAGE at that point).

File: test/reporters/logger.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package reporters
+
+// GinkgoLogger suppresses logging unless "-v" is given to the gingko framework.
+type AzureLogger struct{}
```
Comment (charlesakalugwu) [inline test/reporters/logger.go]: what exactly is `AzureLogger` doing here?

File: test/reporters/logger.go [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+// GinkgoLogger implements Logger.
```
Comment (charlesakalugwu) [inline test/reporters/logger.go]: please delete the commented out code

File: pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml (hunk 1 above)
Comment (mjudeikis) [reply pkg/sync/v5/data/ConfigMap/openshift-azure-logging/fluentd-config.yaml]: image
nill is our new introduced values. <empty> is that any other components puts in into an empty field. I was just trying to follow data consistency

**Pull request merged**